api/repositories/shared.go

```diff
@@ -59,3 +67,73 @@ func getLabelOrAnnotation(mapObj map[string]string, key string) string {
 	}
 	return mapObj[key]
 }
+
+func GetNamespacedResourceNamespace(ctx context.Context, privilegedClient client.Client, userClientFactory UserK8sClientFactory, authInfo authorization.Info, resourceGUID string, resourceList client.ObjectList) (string, error) {
```

Comment:
There are several issues we find with extracting the code into a shared utility:
1. The privileged client is still part of the function signature, i.e. should be still available in the caller. This is one of the things we want to get rid of, i.e. handlers should only use unprivileged clients. We would recommend putting the code into a type that has the privileged client as a private property and inject that type into handlers.
2. The `userClientFactory` is not used, so it makes no sense to be in the function arguments. Moreover, this utility should be using the privileged client only.

Reply:
1. Agreed and we talked about this, but wanted to put the general idea out there for feedback.
2. Agreed - this is a result of moving things around while exploring multiple solutions.